src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockgrid/umbblockgridentries.component.js

```diff
@@ -153,440 +145,158 @@
-                    const fromCtrl = evt.from['Sortable:controller']();
-                    const prevEntries = fromCtrl.entries;
-                    const syncEntry = prevEntries[oldIndex];
-
-                    // Make sure Property Editor Proxies are destroyed, as we need to establish new when moving context:
-
-
-                    // unregister all property editor proxies via events:
-                    fromCtrl.containedPropertyEditorProxies.forEach(slotName => {
-                        _removePropertyProxy(evt.from, slotName);
-                    });
-                    
-                    // Perform the transfer:
-                    if (Sortable.active && Sortable.active.lastPullMode === 'clone') {
-                        syncEntry = Utilities.copy(syncEntry);
-                        prevEntries.splice(Sortable.utils.index(evt.clone, sortable.options.draggable), 0, prevEntries.splice(oldIndex, 1)[0]);
-                    }
-                    else {
-                        prevEntries.splice(oldIndex, 1);
-                    }
-                    vm.entries.splice(newIndex, 0, syncEntry);
-
-                    const contextColumns = vm.blockEditorApi.internal.getContextColumns(vm.parentBlock, vm.areaKey);
-
-                    // if colSpan is lower than contextColumns, and we do have some columnSpanOptions:
-                    if (syncEntry.columnSpan < contextColumns && syncEntry.$block.config.columnSpanOptions.length > 0) {
-                        // then check if the colSpan is a columnSpanOption, if NOT then reset to contextColumns.
-                        const found = syncEntry.$block.config.columnSpanOptions.find(option => option.columnSpan === syncEntry.columnSpan);
-                        if(!found) {
-                            syncEntry.columnSpan = contextColumns;
-                        }
-                    } else {
-                        syncEntry.columnSpan = contextColumns;
-                    }
-                    
-                }
-                else {
-                    vm.entries.splice(newIndex, 0, vm.entries.splice(oldIndex, 1)[0]);
+            // Get grid template:
+            const approvedContainerGridColumns = approvedContainerComputedStyles.gridTemplateColumns.trim().split("px").map(x => Number(x)).filter(n => n > 0).map((n, i, list) => list.length === i ? n : n + gridColumnGap);
+
+            // ensure all columns are there.
+            // This will also ensure handling non-css-grid mode,
+            // use container width divided by amount of columns( or the item width divided by its amount of columnSpan)
+            let amountOfColumnsInWeightMap = approvedContainerGridColumns.length;
+            const amountOfUnknownColumns = gridColumnNumber-amountOfColumnsInWeightMap;
+            if(amountOfUnknownColumns > 0) {
+                let accumulatedValue = getAccumulatedValueOfIndex(amountOfColumnsInWeightMap, approvedContainerGridColumns) || 0;
```

Comment:
The function `getAccumulatedValueOfIndex` does not seem to be present inside the scope of this file.

Reply:
I got around to testing the Flexbox solution and corrected a few details. Works well now.